spacy/cli/project/parallel.py

```diff
@@ -0,0 +1,467 @@
+"""
+Permits the execution of a parallel command group.
+
+The main process starts a worker process for each spaCy projects command in the parallel group.
+Each spaCy projects command consists of n OS level commands. The worker process starts
+subprocesses for these OS level commands serially, and sends status information about their
+execution back to the main process.
+
+The main process maintains a state machine for each spaCy projects command/worker process. The meaning
+of the states is documented alongside the _ParallelCommandInfo.STATES code. Note that the distinction
+between the states 'failed' and 'terminated' is not meaningful on Windows, so that both are displayed
+as 'failed/terminated' on Windows systems.
+"""
+from typing import Any, List, Optional, Dict, Union, cast, Iterator
+import os
+import sys
+import queue
+from pathlib import Path
+from time import time
+from multiprocessing import Manager, Queue, get_context
+from multiprocessing.context import SpawnProcess
+from tempfile import mkdtemp
+from signal import SIGTERM
+from subprocess import STDOUT, Popen, TimeoutExpired
+from dataclasses import dataclass, field
+from wasabi import msg
+from wasabi.util import color, supports_ansi
+
+from .._util import check_rerun, check_deps, update_lockfile, load_project_config
+from ...util import SimpleFrozenDict, working_dir, split_command, join_command
+from ...util import check_bool_env_var, ENV_VARS
+from ...errors import Errors
+
+# Use spawn to create worker processes on all OSs for consistency
+_mp_context = get_context("spawn")
+
+# How often the worker processes managing the commands in a parallel group
+# send keepalive messages to the main process (seconds)
+_PARALLEL_GROUP_STATUS_INTERVAL = 1
+
+# The maximum permissible width of divider text describing a parallel command group
+_MAX_WIDTH_DIVIDER = 60
+
+# Whether or not to display realtime status information
+_DISPLAY_STATUS_TABLE = sys.stdout.isatty() and supports_ansi()
```

Comment:
Unfortunately it looks like `isatty()` doesn't always work correctly on Windows, e.g. with Git Bash: https://stackoverflow.com/questions/40912072/git-for-windows-mintty-sys-stdout-isatty-returns-false

Reply:
Do you have any ideas about how this could be improved? The aim of `isatty()` is to check we're on a console rather than writing to a log file to decide whether or not to use the richer GUI with cursor repositioning. In this link it seems that `isatty()` is returning false negative results, which is not great but I think we can live with: it means a user who could use the richer GUI gets the simpler one. IMHO it would be a bigger problem if `isatty()` returned false positive results because a user would then get the richer GUI in a situation where it wouldn't be rendered properly.

Reply:
It seems this was actually fixed in 2019: https://github.com/mintty/mintty/issues/56?

Reply:
Unfortunately I did run into this issue back in August 2022 so I'd like to keep this comment open for now to potentially explore other alternatives or workarounds, if there are any.

Reply:
Do you remember whether you specifically tested the value of `is_atty()` and found it to be falsely negative, or whether you observed that the whole statement

```
sys.stdout.isatty() and supports_ansi()
```

returned a false-negative result and hence that you were not seeing the richer GUI on Windows? If the second is the case, this will have been because the second condition in the statement, `supports_ansi()`, was returning a false-negative result because of a problem with Wasabi and Colorama that has since been fixed: https://github.com/explosion/wasabi/pull/31. I can confirm that I got the fallback GUI on Windows before this was merged and the rich GUI on Windows after it was merged.

Reply:
Thanks for the context! I'll have to go back and double check with the new version of `wasabi`.